From 5e1ecfae4c251ce8a0e61200a202b48ded466e63 Mon Sep 17 00:00:00 2001
From: connorhartley <vectrixu+gh@gmail.com>
Date: Sun, 24 Mar 2019 11:33:25 +1300
Subject: [PATCH] Add setting for not saving entities in the world


diff --git a/src/main/java/net/minecraft/server/ChunkRegionLoader.java b/src/main/java/net/minecraft/server/ChunkRegionLoader.java
index e6906eff..f8cc33d7 100644
--- a/src/main/java/net/minecraft/server/ChunkRegionLoader.java
+++ b/src/main/java/net/minecraft/server/ChunkRegionLoader.java
@@ -377,6 +377,12 @@ public class ChunkRegionLoader implements IChunkLoader, IAsyncChunkSaver {
                 }
                 // Paper end
 
+                // Paper start
+                if (entity.noSave) {
+                    continue;
+                }
+                // Paper end
+
                 nbttagcompound1 = new NBTTagCompound();
                 if (entity.d(nbttagcompound1)) {
                     chunk.g(true);
diff --git a/src/main/java/net/minecraft/server/Entity.java b/src/main/java/net/minecraft/server/Entity.java
index 74bbba01..1aa910f4 100644
--- a/src/main/java/net/minecraft/server/Entity.java
+++ b/src/main/java/net/minecraft/server/Entity.java
@@ -188,6 +188,7 @@ public abstract class Entity implements ICommandListener, KeyedObject { // Paper
     public boolean fromMobSpawner;
     public boolean spawnedViaMobSpawner; // Paper - Yes this name is similar to above, upstream took the better one
     protected int numCollisions = 0; // Paper
+    public boolean noSave = false; // Paper
     public void inactiveTick() { }
     // Spigot end
 
diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java
index 47edc049..a695a16f 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java
@@ -807,4 +807,16 @@ public abstract class CraftEntity implements org.bukkit.entity.Entity {
         return getHandle().spawnedViaMobSpawner;
     }
     // Paper end
+
+    // Paper start
+    @Override
+    public boolean isNoSave() {
+        return getHandle().noSave;
+    }
+
+    @Override
+    public void setNoSave(boolean noSave) {
+        getHandle().noSave = noSave;
+    }
+    // Paper end
 }
-- 
2.15.1.windows.2

