From cd9dd135f056b3709b9fc56ca4ae69499bb3db3b Mon Sep 17 00:00:00 2001
From: connorhartley <vectrixu+gh@gmail.com>
Date: Sat, 23 Mar 2019 23:08:15 +1300
Subject: [PATCH] Add setting for not saving entities in the world


diff --git a/src/main/java/org/bukkit/entity/Entity.java b/src/main/java/org/bukkit/entity/Entity.java
index e2a2b78c..f554e8ba 100644
--- a/src/main/java/org/bukkit/entity/Entity.java
+++ b/src/main/java/org/bukkit/entity/Entity.java
@@ -523,4 +523,20 @@ public interface Entity extends Metadatable, CommandSender, Nameable {
      */
     Chunk getChunk();
     // Paper end
+
+    // Paper start
+    /**
+     * Returns whether this entity will be saved in the world.
+     *
+     * @return True if the entity is not going to be saved in the world
+     */
+    boolean isNoSave();
+
+    /**
+     * Sets whether this entity will be saved in the world.
+     *
+     * @param noSave whether to save in the world
+     */
+    void setNoSave(boolean noSave);
+    // Paper end
 }
-- 
2.15.1.windows.2

